job/cache.go

```diff
@@ -115,6 +115,18 @@ func (c *MemoryJobCache) Delete(id string) {
 	c.jobs.Lock.Lock()
 	defer c.jobs.Lock.Unlock()
 
+	j := c.jobs.Jobs[id]
+	if j == nil {
+		// TODO
+		//return nil, JobDoesntExistErr
+		return
+	}
+
+	// TODO return error
+	c.jobs.Lock.Unlock()
```

Comment:
@ajvb There are almost certainly too many locks in this program. Think about how you can use channels to simplify all this; it's getting too hard to get right IMO.